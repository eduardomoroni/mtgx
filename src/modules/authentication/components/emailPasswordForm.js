// @flow

import React, { Component } from 'react'
import { PropTypes } from 'prop-types'
import I18n from 'react-native-i18n'
import { View } from 'react-native'
import { Input, Button } from 'nachos-ui'
import { styles } from './styles/emailPasswordForm.style'

export class EmailPasswordForm extends Component {
  state = {
    email: '',
    password: ''
  }

  static propTypes = {
    buttonText: PropTypes.string.isRequired,
    buttonCallback: PropTypes.func.isRequired
  }

  render () {
    const { buttonCallback, buttonText } = this.props
    const { email, password } = this.state

    return (
      <View>
        <Input
          icon='ios-mail-outline'
          style={[styles.input]}
          placeholder={I18n.t('EMAIL')}
          value={this.state.email}
          onChangeText={email => this.setState({email})}
        />
        <Input
          icon='ios-lock-outline'
          style={[styles.input]}
          placeholder={I18n.t('PASSWORD')}
          value={this.state.password}
          onChangeText={password => this.setState({password})}
        />
        <Button
          kind='squared'
          type='success'
          style={[styles.button]}
          onPress={() => buttonCallback(email, password)}
        >
          {buttonText}
        </Button>
      </View>
    )
  }
}
