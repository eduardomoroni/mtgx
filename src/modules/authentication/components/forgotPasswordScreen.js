// @flow

import React, { Component } from 'react'
import I18n from 'react-native-i18n'
import PropTypes from 'prop-types'
import { View } from 'react-native'
import { H1, Button, Input } from 'nachos-ui'
import { styles } from './styles/forgotPassword.style'

export class ForgotPasswordScreen extends Component {
  state = {
    email: ''
  }

  static propTypes = {
    resetPassword: PropTypes.func.isRequired
  }

  onForgotPasswordPressed = async () => {
    this.props.resetPassword(this.state.email)
  }

  render () {
    return (
      <View style={styles.container}>
        <H1>{I18n.t('RESET_PASSWORD')}</H1>
        <Input
          icon='ios-mail-outline'
          style={[styles.input]}
          placeholder={I18n.t('EMAIL')}
          value={this.state.email}
          onChangeText={email => this.setState({email})}
        />
        <Button
          kind='squared'
          type='success'
          style={[styles.button]}
          onPress={this.onForgotPasswordPressed}
        >
          {I18n.t('RESET_PASSWORD')}
        </Button>
      </View>
    )
  }
}
