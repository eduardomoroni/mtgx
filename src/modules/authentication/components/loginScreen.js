// @flow

import React, { Component } from 'react'
import Config from 'react-native-config'
import I18n from 'react-native-i18n'
import { StyleSheet, Text, View } from 'react-native'
import { H4, Button } from 'nachos-ui'

export class LoginScreen extends Component {
  renderButton () {
    const btnStyle = { margin: 15 }
    return (
      <View>
        <H4>Example:</H4>
        <Button kind='squared' type='success' style={btnStyle}>
          Success
        </Button>
        <Button kind='squared' type='danger' style={btnStyle}>
          Danger
        </Button>
        <Button
          kind='squared'
          iconName='md-cloud-download'
          style={btnStyle}
        >
          Primary
        </Button>
        <H4 align='center'>Disabled style</H4>
      </View>
    )
  }

  render () {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          Welcome to React Native!
        </Text>
        <Text style={styles.instructions}>
          To get started, edit index.ios.js
        </Text>
        <Text style={styles.instructions}>
          Press Cmd+R to reload,{'\n'}
          Cmd+D or shake for dev menu
        </Text>
        <Text style={styles.instructions}>
          {Config.TEST_ENV}
        </Text>
        <Text style={styles.instructions}>
          {I18n.t('test')}
        </Text>
        {this.renderButton()}
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF'
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5
  }
})
