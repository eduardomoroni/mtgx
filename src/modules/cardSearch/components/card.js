// @flow

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View, Text } from 'react-native'

import { placeholdersToSymbols } from '../../../services/realm/conversion/placeholder' // TODO: REVISIT
import { cardType, printingsType } from '../types/cardType'
import { styles } from './styles/card.styles'

// TODO: REVISIT
export const getLastPrinting = (printings: printingsType) => {
  const keys = Object.keys(printings)
  const lastPrinting = printings[keys[keys.length - 1]]
  return lastPrinting.printing
}

export class Card extends Component {
  static propTypes = {
    showCardText: PropTypes.bool,
    card: cardType
  }

  renderCardNameAndMana = () => {
    const { name, manaCost } = this.props.card

    return (
      <View style={styles.lineContainer}>
        <View style={styles.leftWord}>
          <Text key='cardName' style={styles.cardNameText}>
            {name}
          </Text>
        </View>
        <View style={styles.rightWord}>
          <Text key='cardManaCost' style={styles.mana}>
            {placeholdersToSymbols(manaCost)}
          </Text>
        </View>
      </View>
    )
  }

  renderCardTypeAndEdition = () => {
    const { type, printings } = this.props.card

    return (
      <View style={styles.lineContainer}>
        <View style={styles.leftWord}>
          <Text key='cardType' style={styles.cardTypeText}>
            {type}
          </Text>
        </View>
        <View style={styles.rightWord}>
          <Text key='cardEdition'>
            {getLastPrinting(printings)}
          </Text>
        </View>
      </View>
    )
  }

  renderTextAndPower = () => {
    const { card, showCardText } = this.props
    const { text, power, toughness } = card

    return (
      <View style={styles.cardTextContainer}>
        <View style={styles.cardText}>
          <Text key='cardText' style={styles.text}>
            {showCardText ? text : ''}
          </Text>
        </View>
        <View style={styles.cardPower}>
          <Text key='cardPowerAndToughness' style={styles.cardPowerToughness}>
            {power || toughness ? `${power}/${toughness}` : ''}
          </Text>
        </View>
      </View>
    )
  }

  render () {
    return (
      <View style={styles.container}>
        {this.renderCardNameAndMana()}
        {this.renderCardTypeAndEdition()}
        {this.renderTextAndPower()}
      </View>
    )
  }
}
