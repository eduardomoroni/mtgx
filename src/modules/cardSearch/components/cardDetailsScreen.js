// @flow

import I18n from 'react-native-i18n'
import React, { Component } from 'react'
import { Text, View, ScrollView } from 'react-native'

import { styles } from './styles/cardDetailsScreen.styles'
import { CardImage } from './cardImage'
import { cardType } from '../types/cardType'

export class CardDetailsScreen extends Component {
  static propTypes = {
    card: cardType
  }

  renderRow = (rowName: string, rowText: string) => {
    if (rowText === null) {
      return null
    }

    return (
      <View style={styles.rowContainer}>
        <Text style={styles.rowText}>
          <Text style={styles.rowName}>{`${I18n.t(rowName)}: `}</Text>
          {rowText}
        </Text>
      </View>
    )
  }

  render () {
    const { card } = this.props

    return (
      <ScrollView style={styles.container}>
        <CardImage style={styles.card} multiverseId={card.multiverseid} />
        {this.renderRow('CARD_NAME', card.name)}
        {this.renderRow('TYPE', card.type)}
        {this.renderRow('POWER_TOUGHNESS', `${card.power}/${card.toughness}`)}
        {this.renderRow('MANA_COST', `${card.manaCost} (${card.cmc})`)}
        {this.renderRow('RARITY', card.rarity)}
        {this.renderRow('CARD_TEXT', card.text)}
        {this.renderRow('FLAVOR_TEXT', card.flavor)}
      </ScrollView>
    )
  }
}
