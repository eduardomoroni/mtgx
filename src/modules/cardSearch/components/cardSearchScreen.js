import React, { Component } from 'react'
import { string, arrayOf, func } from 'prop-types'
import { View } from 'react-native'
import { ManaSymbolBar } from './manaSymbolBar'
import { Field } from 'redux-form/immutable'
import Modal from 'react-native-modal'

import {
  TextInputForm,
  NumericInputForm,
  DropdownInputForm,
  ModalToggle,
  MultiSelect
} from '../../form/components'

import { styles } from './styles/cardSearch.styles'

export class CardSearchScreen extends Component {
  static propTypes = {
    type: string,
    rarity: string,
    format: string,
    subType: string,
    sets: arrayOf(string),
    types: arrayOf(string),
    colors: arrayOf(string),
    formats: arrayOf(string),
    rarities: arrayOf(string),
    subtypes: arrayOf(string),
    colorsIdentity: arrayOf(string),
    showModal: func
  }

  openModal = (fieldName, items) => {
    // TODO: wix/react-native-navigation has built-in Modal, Why are we using an external one?
    return (
      <Modal isVisible onModalHide={() => this.props.showModal('')} >
        <Field name={fieldName} component={MultiSelect} items={items} />
      </Modal>
    )
  }

  switchModals = () => {
    const { visibleModal, formats, sets, rarities } = this.props

    switch (visibleModal) {
      case 'cardRarity':
        return this.openModal('cardRarity', rarities)
      case 'cardSet':
        return this.openModal('cardSet', sets)
      case 'cardFormat':
        return this.openModal('cardFormat', formats)
      default:
        return <View />
    }
  }

  renderThreeColumnsLine = (fieldOne, fieldTwo, fieldThree) => {
    return (
      <View style={styles.multipleFieldsPerLine}>
        <View style={styles.leftField}>{ fieldOne }</View>
        <View style={styles.middleField}>{ fieldTwo }</View>
        <View style={styles.rightField}>{ fieldThree }</View>
      </View>
    )
  }

  renderTwoColumnsLine = (firstColumn, secondColumn) => {
    return (
      <View style={styles.multipleFieldsPerLine}>
        <View style={styles.leftField}>{ firstColumn }</View>
        <View style={styles.rightField}>{ secondColumn }</View>
      </View>
    )
  }

  render () {
    const {
      set,
      type,
      types,
      rarity,
      format,
      colors,
      subType,
      subTypes,
      showModal,
      colorsIdentity
    } = this.props

    return (
      <View style={styles.container}>
        <View style={styles.formContainer}>
          {/* TODO: Add selectedColor funcionality */}
          <Field name='COLORS' component={ManaSymbolBar} selectedColor={colors} />
          <Field name='COLOR_IDENTITY' component={ManaSymbolBar} selectedColor={colorsIdentity} />
          <Field name='CARD_NAME' component={TextInputForm} />
          <Field name='CARD_TEXT' component={TextInputForm} />
          <Field name='FLAVOR_TEXT' component={TextInputForm} />
          {this.renderTwoColumnsLine(
            <Field name='TYPE' component={DropdownInputForm} dropdownItems={types} selectedValue={type} />,
            <Field name='SUB_TYPE' component={DropdownInputForm} dropdownItems={subTypes} selectedValue={subType} />
          )}
          {this.renderTwoColumnsLine(
            <Field name='ARTIST' component={TextInputForm} />,
            <Field name='COLLECTION_NUMBER' component={TextInputForm} keyboardType={'numeric'} maxLength={3} />
          )}
          {this.renderThreeColumnsLine(
            <Field name='POWER' component={NumericInputForm} />,
            <Field name='TOUGHNESS' component={NumericInputForm} />,
            <Field name='CMC' component={NumericInputForm} />
          )}
          {/* TODO: Implement Modal */}
          {this.renderThreeColumnsLine(
            <ModalToggle label='RARITY' onPress={() => showModal} selected={rarity} />,
            <ModalToggle label='SET' onPress={() => showModal} selected={set} />,
            <ModalToggle label='FORMAT' onPress={() => showModal} selected={format} />
          )}
        </View>
        <View style={styles.containerFooter} />
        {this.switchModals()}
      </View>
    )
  }
}
