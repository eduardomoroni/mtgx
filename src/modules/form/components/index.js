// @flow

export * from './inputLabel'
export * from './inputPicker'
export * from './textInputForm'
export * from './switchInputForm'
export * from './modalToggle'
export * from './multiSelect'
// $FlowFixMe
export * from './numericInputForm'
// $FlowFixMe
export * from './dropdownInputForm'
