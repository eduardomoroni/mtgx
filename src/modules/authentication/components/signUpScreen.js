// @flow

import React, { Component } from 'react'
import I18n from 'react-native-i18n'
import PropTypes from 'prop-types'
import { View } from 'react-native'
import { H1 } from 'nachos-ui'
import { styles } from './styles/loginScreen.style'
import { EmailPasswordForm } from './emailPasswordForm'

export class SignUpScreen extends Component {
  static propTypes = {
    signUpUser: PropTypes.func.isRequired
  }

  onSignUpPressed = async (email: string, password: string) => {
    this.props.signUpUser(email, password)
  }

  render () {
    return (
      <View style={styles.container}>
        <H1>{I18n.t('SIGNUP')}</H1>
        <EmailPasswordForm
          buttonText={I18n.t('SIGNUP')}
          buttonCallback={this.onSignUpPressed}
        />
      </View>
    )
  }
}
