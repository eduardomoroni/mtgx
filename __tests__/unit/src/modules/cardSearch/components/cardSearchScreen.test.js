import React from 'react'
import { shallow } from 'enzyme'
import Modal from 'react-native-modal'
import { Field } from 'redux-form/immutable'
import { Button } from 'nachos-ui'
import { CardSearchScreen, cardSearchModals } from '../../../../../../src/modules/cardSearch/components/cardSearchScreen'

const mockedForm = {}
const props = {
  sets: ['set'],
  types: ['type'],
  formats: ['format'],
  rarities: ['rarity'],
  subTypes: ['subType'],
  submitCardSearchForm: jest.fn(() => Promise.resolve()),
  handleSubmit: jest.fn(callback => callback(mockedForm)),
  navigator: {
    push: jest.fn()
  }
}

describe('<CardSearchScreen />', () => {
  const wrapper = shallow(<CardSearchScreen {...props} />)
  const modalId = 'RARITY'

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('should have a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should show search modal', () => {
    wrapper.setState({ visibleModal: cardSearchModals[modalId] })
    const modal = wrapper.find(Modal)

    expect(modal.exists()).toBe(true)
    expect(modal.find(Field).prop('name')).toEqual(modalId)
  })

  it('should toggle visible modal', () => {
    const { showModal, hideModal } = wrapper.instance()

    showModal(modalId)()
    expect(wrapper.state('visibleModal')).toEqual(cardSearchModals[modalId])

    hideModal()
    expect(wrapper.state('visibleModal')).toEqual('')
  })

  it('should search for cards and navigate to cards.results screen', async () => {
    const button = wrapper.find(Button)
    await button.simulate('press')

    expect(props.submitCardSearchForm).toHaveBeenCalledWith(mockedForm)
    expect(props.navigator.push).toHaveBeenCalledWith({screen: 'card.results'})
  })

  it('should not navigate in case of exception during card search', async () => {
    const mockError = jest.fn(() => Promise.reject(new Error('TEST ERROR - JUST IGNORE')))
    const wrapper = shallow(<CardSearchScreen {...props} submitCardSearchForm={mockError} />)

    await wrapper.find(Button).simulate('press')
    expect(props.navigator.push).toHaveBeenCalledTimes(0)
  })
})
