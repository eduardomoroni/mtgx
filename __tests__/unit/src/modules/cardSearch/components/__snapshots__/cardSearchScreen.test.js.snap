// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardSearchScreen /> Should have a snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "marginHorizontal": 10,
        "paddingTop": 69,
      }
    }
  >
    <Field
      component={[Function]}
      name="COLORS"
    />
    <Field
      component={[Function]}
      name="COLOR_IDENTITY"
    />
    <Field
      component={[Function]}
      name="CARD_NAME"
    />
    <Field
      component={[Function]}
      name="CARD_TEXT"
    />
    <Field
      component={[Function]}
      name="FLAVOR_TEXT"
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-start",
            "marginRight": 1.6666666666666667,
          }
        }
      >
        <Field
          component={[Function]}
          name="TYPE"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
            "marginLeft": 1.6666666666666667,
          }
        }
      >
        <Field
          component={[Function]}
          name="SUB_TYPE"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-start",
            "marginRight": 1.6666666666666667,
          }
        }
      >
        <Field
          component={[Function]}
          name="ARTIST"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
            "marginLeft": 1.6666666666666667,
          }
        }
      >
        <Field
          component={[Function]}
          keyboardType="numeric"
          maxLength={3}
          name="COLLECTION_NUMBER"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-start",
            "marginRight": 1.6666666666666667,
          }
        }
      >
        <Field
          component={[Function]}
          name="POWER"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "marginLeft": 1.6666666666666667,
          }
        }
      >
        <Field
          component={[Function]}
          name="TOUGHNESS"
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
            "marginLeft": 1.6666666666666667,
          }
        }
      >
        <Field
          component={[Function]}
          name="CMC"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-start",
            "marginRight": 1.6666666666666667,
          }
        }
      >
        <ModalToggle
          label="RARITY"
          onPress={[Function]}
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "marginLeft": 1.6666666666666667,
          }
        }
      >
        <ModalToggle
          label="SET"
          onPress={[Function]}
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
            "marginLeft": 1.6666666666666667,
          }
        }
      >
        <ModalToggle
          label="FORMAT"
          onPress={[Function]}
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
      }
    }
  />
  <View />
</View>
`;
